### ISSUE TITLE
HIGHSOURCE PROJECT STATUS UPDATE

### ISSUE THREAD

**Issue by mattrpav** [highsource/maven-jaxb2-plugin#206]
I have confirmed with Aleksei's wife that he passed away last January. My thoughts were that it would be best to continue updates under the 'highsource' org. Give Aleksei a solid for the work he did over the years, and provide consistency to users.

I opened a death transfer request with GitHub 2 weeks ago to gain some level of access to the repositories to continue Aleksei's work out under 'highsource'. GitHub's support team is notably behind and has not responded despite several requests for updates.

I plan to give GitHub another few days to respond before looking to next steps. 

About me-- I am a long time open source contributor and Apache committer. Professionally, I head up the technical side at HYTE and we have resources we can put to updating the Maven JAXB2 Plugin and jaxb2-basics.

Message from his wife:
```
Aleksei passed away on 16.1.2021. 
 
https://good-bye.org/alexey-valikov/
 
He would be so proud to know that his projects live after his death. 
 
If you can overtake some projects, that would be great. 
```

Thanks,
Matt

**Comment by jedvardsson** [comment]
Oh. I am sorry to hear about Aleksei. Rest in piece.

Great initiative, @mattrpav. Thumbs up.

**Comment by patrodyne** [comment]
My condolences to Aleksei's wife and family. I have forked `maven-jaxb2-plugin` to a new repository called [hisrc-higherjaxb](https://github.com/patrodyne/hisrc-higherjaxb#readme). I have published a release to Maven Central for this plugin. Also, I have forked several other related Highsource repositories and published releases for them. My goal for these releases is to resolve warnings and errors that have accrued over time. The new releases have been built using Java 11 but are source and target compatible with Java 8 and up.

**Comment by vetrivendhan** [comment]
Hi @mattrpav 
Do you know when can we expect the jaxb2-basics should be made compatible with Jakarta JAXB? or can you point me if there any similar plugins for me to achieve this?

Thanks in advance!!

**Comment by vetrivendhan** [comment]
Thank for your swift response @mattrpav. Please let me know once it is released to Maven central.

**Comment by vetrivendhan** [comment]
Hi, @mattrpav Do you have any updates on it?

**Comment by ssill2** [comment]
@mattrpav Thanks for taking this on.  I have several applications that make use of this plugin and I just started looking at what it's going to take to go from java11 to java 17.  The code generations stuff, as expected, failed right out of the gate lol.  This is why I start well ahead of   java 11 EOL.  I'm following your progress with bated breath! :)   Let me know if there's anything I can do.

**Comment by vetrivendhan** [comment]
Hi @ssill2 When I use for Java 11+ below error is thrown in the generated class.. Can you please assist me ?

JAXBEqualsStrategy.getInstance(); -- Compilation fails

**Comment by ssill2** [comment]
This has always been a fickle sort of thing.
The big thing when I switched from java 8 to java 11 was adding
`        
        <dependency>
            <groupId>javax.xml.bind</groupId>
            <artifactId>jaxb-api</artifactId>
            <version>2.3.1</version>
        </dependency>
        <dependency>
            <groupId>com.sun.xml.bind</groupId>
            <artifactId>jaxb-core</artifactId>
            <version>2.3.0.1</version>
        </dependency>
        <dependency>
            <groupId>com.sun.xml.bind</groupId>
            <artifactId>jaxb-impl</artifactId>
            <version>2.3.1</version>
        </dependency>`

And then for the actual invocation of the plugin
`            <plugin>
                <groupId>org.jvnet.jaxb2.maven2</groupId>
                <artifactId>maven-jaxb23-plugin</artifactId>
                <version>0.14.0</version>
                <configuration>                    
                    <outputDirectory>${basedir}/target/generated-sources/jaxb</outputDirectory>
                    <args>
                        <arg>-XautoNameResolution</arg>
                        <arg>-Xannotate</arg>
                        <arg>-Xnamespace-prefix</arg>
                    </args>
                    <enableIntrospection>true</enableIntrospection>                           
                    <verbose>true</verbose>
                    <extension>true</extension>
                    <catalog>${basedir}/src/main/catalogs/schema_catalog.xml</catalog>
                    <clearOutputDir>false</clearOutputDir>
                    <strict>false</strict>
                    <accessExternalSchema>all</accessExternalSchema>
                    <accessExternalDTD>all</accessExternalDTD>
                </configuration>
                <executions>
                    <execution>
                        <id>compile_taxii_schemas</id>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <bindingDirectory>${basedir}/src/main/xjb</bindingDirectory>
                            <schemaDirectory>${basedir}/src/main/xsd</schemaDirectory>
                            <schemaIncludes>
                                <include>external/xmldsig/*.xsd</include>
                                <include>taxii_1.1/*.xsd</include>
                                <include>stix_1.2/indicator.xsd</include>
                                <include>stix_1.2/stix_common.xsd</include>
                                <include>stix_1.2/stix_core.xsd</include>
                                <include>stix_1.2/data_marking.xsd</include>
                                <include>stix_1.2/threat_actor.xsd</include>
                                <include>stix_1.2/cybox/cybox_common.xsd</include>
                                <include>stix_1.2/cybox/cybox_core.xsd</include>
                                <include>stix_1.2/cybox/cybox_default_vocabularies.xsd</include>
                                <include>stix_1.2/cybox/objects/Address_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/DNS_Query_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/DNS_Record_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/File_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Connection_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Flow_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Packet_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Route_Entry_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Route_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Socket_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Subnet_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Port_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/URI_Object.xsd</include>
                                <include>stix_1.2/ttp.xsd</include>
                            </schemaIncludes>
                            <schemaExcludes>
                            </schemaExcludes>
                        </configuration>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>org.jvnet.jaxb2_commons</groupId>
                        <artifactId>jaxb2-basics-annotate</artifactId>
                        <version>1.1.0</version>
                        <scope>compile</scope>
                    </dependency>
                    <dependency>
                        <groupId>org.jvnet.jaxb2_commons</groupId>
                        <artifactId>jaxb2-basics</artifactId>
                        <version>1.11.1</version>
                        <scope>compile</scope>
                    </dependency>
                    <dependency>
                        <groupId>org.jvnet.jaxb2_commons</groupId>
                        <artifactId>jaxb2-namespace-prefix</artifactId>
                        <version>1.3</version>
                        <scope>compile</scope>
                    </dependency>
                </dependencies>
            </plugin>
`

**Comment by ssill2** [comment]
The above works perfectly for me in java 11.

**Comment by ssill2** [comment]
Of course you also need to be using a newer maven, 3.6.3+, I use 3.8.x,  and use a later version of maven-compiler-plugin
I use
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.10.1</version>
                <configuration>
                    <!-- put your configurations here -->
                    <release>11</release>
                    <debug>false</debug>
                    <debuglevel>lines,vars,source</debuglevel>
                </configuration>
            </plugin>
```

**Comment by vetrivendhan** [comment]
Thanks for your response @ssill2 will this work for the below arguments as well

                                <arg>-XtoString</arg>
                                <arg>-Xequals</arg>
                                <arg>-XhashCode</arg>
                                <arg>-Xcopyable</arg>
                                <arg>-Xmergeable</arg>

**Comment by ssill2** [comment]
I'm not sure about that.  I think the key is to add any dependencies to plugin that you reference with those -X args.

**Comment by mattrpav** [comment]
@vetrivendhan @ssill2 I have the build passing all but 2 unit tests. Not sure if they were working before. I'll publish a test release with those tests disabled (create tickets for them) and then go from there.

**Comment by ssill2** [comment]
sounds great! thanks matt!

**Comment by vetrivendhan** [comment]
@mattrpav Looking forward to your release... Thanks a lot!! Let me know once it is done

**Comment by ssill2** [comment]
@mattrpav you guys need to establish a tip jar lol, I'd be happy to drop in a few bucks as I'm sure all the other folks that use this plugin would too.

**Comment by ssill2** [comment]
After several hours of messing around with the particular project that does my schema generation I was able to make a fork of this project, https://github.com/evolvedbinary/jvnet-jaxb-maven-plugin, work under java 17.

**Comment by mattrpav** [comment]
@ssill2 can you make a PR against the highsource tree?

**Comment by ssill2** [comment]
The evolvedbinary fork isn't mine, I saw a reference to it when I was fumbling around today.   Would the creator of that fork be the one to create the PR?  

 0.15.0 of their plugin is published to maven central repo.  And miraculously this all works compiling with java 11 and with a module-info.java in the project to boot.  

Basically here's the meat of the changes to my project

replace the javax.xml.bind:jaxb-api and any com.sun.xml:jaxb-impl dependencies with
```
        <dependency>
            <groupId>jakarta.xml.bind</groupId>
            <artifactId>jakarta.xml.bind-api</artifactId>
            <version>3.0.1</version>
        </dependency>

```

And the plugin to do the work
```
            <plugin>
                <groupId>com.evolvedbinary.maven.jvnet</groupId>
                <artifactId>jaxb30-maven-plugin</artifactId>
                <version>0.15.0</version>
                <configuration>
                    <outputDirectory>${basedir}/target/generated-sources/jaxb</outputDirectory>
                    <args>
                        <arg>-XautoNameResolution</arg>
                        <arg>-Xannotate</arg>
                    </args>
                    <enableIntrospection>true</enableIntrospection>
                    <verbose>true</verbose>
                    <extension>true</extension>
                    <catalog>${basedir}/src/main/catalogs/schema_catalog.xml</catalog>
                    <clearOutputDir>false</clearOutputDir>
                    <strict>false</strict>
                    <accessExternalSchema>all</accessExternalSchema>
                    <accessExternalDTD>all</accessExternalDTD>
<!--                    <packageLevelAnnotations>true</packageLevelAnnotations>-->
                </configuration>
                <executions>
                    <execution>
                        <id>compile_taxii_schemas</id>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <bindingDirectory>${basedir}/src/main/xjb</bindingDirectory>
                            <schemaDirectory>${basedir}/src/main/xsd</schemaDirectory>
                            <schemaIncludes>
                                <include>external/xmldsig/*.xsd</include>
                                <include>taxii_1.1/*.xsd</include>
                                <include>stix_1.2/cybox/cybox_common.xsd</include>
                                <include>stix_1.2/cybox/cybox_default_vocabularies.xsd</include>
                                <include>stix_1.2/cybox/cybox_core.xsd</include>
                                <include>stix_1.2/stix_common.xsd</include>
                                <include>stix_1.2/stix_core.xsd</include>
                                <include>stix_1.2/data_marking.xsd</include>
                                <include>stix_1.2/threat_actor.xsd</include>
                                <include>stix_1.2/indicator.xsd</include>
                                <include>stix_1.2/cybox/objects/Address_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/DNS_Query_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/DNS_Record_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/File_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Connection_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Flow_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Packet_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Route_Entry_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Route_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Socket_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Network_Subnet_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/Port_Object.xsd</include>
                                <include>stix_1.2/cybox/objects/URI_Object.xsd</include>
                                <include>stix_1.2/ttp.xsd</include>
                            </schemaIncludes>
                            <schemaExcludes>
                            </schemaExcludes>
                        </configuration>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>jakarta.xml.bind</groupId>
                        <artifactId>jakarta.xml.bind-api</artifactId>
                        <version>3.0.1</version>
                    </dependency>
                    <dependency>
                        <groupId>javax.xml.bind</groupId>
                        <artifactId>jaxb-api</artifactId>
                        <version>2.3.1</version>
                    </dependency>
                    <dependency>
                        <groupId>org.jvnet.jaxb2_commons</groupId>
                        <artifactId>jaxb2-basics-annotate</artifactId>
                        <version>1.1.0</version>
                        <scope>compile</scope>
                    </dependency>
                </dependencies>
            </plugin>

```

**Comment by ssill2** [comment]
to make unit tests pass on that project I need this dep as well as the jakarta.xml.bind:jakarta.xml.bind-api
```
        <dependency>
            <groupId>com.sun.xml.bind</groupId>
            <artifactId>jaxb-impl</artifactId>
            <version>3.0.2</version>
        </dependency>

```

**Comment by vetrivendhan** [comment]
Hi @ssill2 Sources are generating against the XSD with Jakarta package conventions as per the `<plugin>
                <groupId>com.evolvedbinary.maven.jvnet</groupId>
                <artifactId>jaxb30-maven-plugin</artifactId>
                <version>0.15.0</version>`
But the `                            <args>
                                <arg>-XtoString</arg>
                                <arg>-Xequals</arg>
                                <arg>-XhashCode</arg>
                                <arg>-Xcopyable</arg>
                                <arg>-Xmergeable</arg>
                            </args>` these parameters are not working since there is no compatible jakarta based plugin.  Can you please assist me on this regard?

**Comment by vetrivendhan** [comment]
> @vetrivendhan @ssill2 I have the build passing all but 2 unit tests. Not sure if they were working before. I'll publish a test release with those tests disabled (create tickets for them) and then go from there.

Hi @mattrpav Can you please let me know how to get your latest changes?

**Comment by ssill2** [comment]
@vetrivendhan 
I was able to to get this to work even with jakarta.  I had to change my .xjb file to add XmlRootElement from the jakarta package instead of the javax one.

Here's a snippet from my .xjb

```
<?xml version="1.0" encoding="UTF-8"?>
<jaxb:bindings xmlns="https://jakarta.ee/xml/ns/jaxb"
               xmlns:jaxb="https://jakarta.ee/xml/ns/jaxb"               
               xmlns:xsd="http://www.w3.org/2001/XMLSchema"
               xmlns:annox="http://annox.dev.java.net"
               version="3.0">
    
    <jaxb:bindings schemaLocation="../xsd/taxii_1.1/TAXII_XMLMessageBinding_Schema.xsd" node="/xsd:schema">
        <jaxb:schemaBindings>
            <jaxb:package name="org.mitre.taxii.messages.xml11"/>
        </jaxb:schemaBindings>
    </jaxb:bindings>
    
    <jaxb:bindings schemaLocation="../xsd/taxii_1.1/TAXII_DefaultQuery_Schema.xsd" node="/xsd:schema">
        <jaxb:schemaBindings>
            <jaxb:package name="org.mitre.taxii.query.xml11"/>
        </jaxb:schemaBindings>
    </jaxb:bindings>
    
    <jaxb:bindings schemaLocation="../xsd/stix_1.2/stix_core.xsd" node="/xsd:schema" jaxb:extensionBindingPrefixes="annox">
        <jaxb:schemaBindings>
            <jaxb:package name="org.mitre.stix.v1_2.core"/>
        </jaxb:schemaBindings>
        
        <bindings node="//xsd:complexType[@name='STIXType']">
            <class name="STIXPackage"/>
            <annox:annotate>
              <annox:annotate annox:class="jakarta.xml.bind.annotation.XmlRootElement" name="STIX_Package"/>
            </annox:annotate>            
        </bindings>        
    </jaxb:bindings>
```
> Hi @ssill2 Sources are generating against the XSD with Jakarta package conventions as per the `<plugin> <groupId>com.evolvedbinary.maven.jvnet</groupId> <artifactId>jaxb30-maven-plugin</artifactId> <version>0.15.0</version>` But the ` <args> <arg>-XtoString</arg> <arg>-Xequals</arg> <arg>-XhashCode</arg> <arg>-Xcopyable</arg> <arg>-Xmergeable</arg> </args>` these parameters are not working since there is no compatible jakarta based plugin. Can you please assist me on this regard?
> 
> Reference: [davidmoten/jax-maven-plugin#36](https://github.com/davidmoten/jax-maven-plugin/issues/36)

**Comment by ssill2** [comment]
@vetrivendhan 
I have not used the plugins you've mentioned so I can't help you.  Were those also highsource plugins?

> Hi @ssill2 Sources are generating against the XSD with Jakarta package conventions as per the `<plugin> <groupId>com.evolvedbinary.maven.jvnet</groupId> <artifactId>jaxb30-maven-plugin</artifactId> <version>0.15.0</version>` But the ` <args> <arg>-XtoString</arg> <arg>-Xequals</arg> <arg>-XhashCode</arg> <arg>-Xcopyable</arg> <arg>-Xmergeable</arg> </args>` these parameters are not working since there is no compatible jakarta based plugin. Can you please assist me on this regard?
> 
> Reference: [davidmoten/jax-maven-plugin#36](https://github.com/davidmoten/jax-maven-plugin/issues/36)

**Comment by vetrivendhan** [comment]
> @vetrivendhan @ssill2 I have the build passing all but 2 unit tests. Not sure if they were working before. I'll publish a test release with those tests disabled (create tickets for them) and then go from there.

Hi @mattrpav When will your test build will be ready?

**Comment by mattrpav** [comment]
I've updated the release plans:
v0.0.15.1 <-- test release to verify Maven Central accepts the release
v0.15.1 <-- Should be EXACTLY the same as v0.0.15.1 (this is to provide changes for folks waiting on unreleased changes that were targeted for v0.14.1)
v2.3.6 <-- Should be EXACTLY the same as v0.15.1 (to match the JAXB version)

I will then make a v2.x branch to maintain any needed jdk 8.x+jaxb 2.x changes going forward

Then 'master' will be bumped to 'v3.x' (to match JAXB version) and converted to jdk 11+jakarta.

**Comment by ssill2** [comment]
@mattrpav Glad to see you're working on that!   I have something that works in that evolved binary fork of this project, but I'll be happy to test again when v3.x is released.

> I've updated the release plans: v0.0.15.1 <-- test release to verify Maven Central accepts the release (in-progress 2022/04/21) v0.15.1 <-- Should be EXACTLY the same as v0.0.15.1 (this is to provide changes for folks waiting on unreleased changes that were targeted for v0.14.1) v2.3.6 <-- Should be EXACTLY the same as v0.15.1 (to match the JAXB version)
> 
> I will then make a v2.x branch to maintain any needed jdk 8.x+jaxb 2.x changes going forward
> 
> Then 'master' will be bumped to 'v3.x' (to match JAXB version) and converted to jdk 11+jakarta.

**Comment by norrisjeremy** [comment]
> I've updated the release plans: v0.0.15.1 <-- test release to verify Maven Central accepts the release (in-progress 2022/04/21) v0.15.1 <-- Should be EXACTLY the same as v0.0.15.1 (this is to provide changes for folks waiting on unreleased changes that were targeted for v0.14.1) v2.3.6 <-- Should be EXACTLY the same as v0.15.1 (to match the JAXB version)
> 
> I will then make a v2.x branch to maintain any needed jdk 8.x+jaxb 2.x changes going forward
> 
> Then 'master' will be bumped to 'v3.x' (to match JAXB version) and converted to jdk 11+jakarta.

JAXB 3.x works fine with Java 8, I hope you won't force the 3.x update of this plugin require Java 11+?

**Comment by ssill2** [comment]
> that would be a good idea. Highsource has a lot of projects to carry forward, used by a lot of folks. Do you know of a good app or service tied to GitHub that can support open source donations?

I'm not sure, but I'd be happy to contribute if you set something up.  I'll poke around.

**Comment by mattrpav** [comment]
> > I've updated the release plans: v0.0.15.1 <-- test release to verify Maven Central accepts the release (in-progress 2022/04/21) v0.15.1 <-- Should be EXACTLY the same as v0.0.15.1 (this is to provide changes for folks waiting on unreleased changes that were targeted for v0.14.1) v2.3.6 <-- Should be EXACTLY the same as v0.15.1 (to match the JAXB version)
> > I will then make a v2.x branch to maintain any needed jdk 8.x+jaxb 2.x changes going forward
> > Then 'master' will be bumped to 'v3.x' (to match JAXB version) and converted to jdk 11+jakarta.
> 
> JAXB 3.x works fine with Java 8, I hope you won't force the 3.x update of this plugin require Java 11+?

Running two JAXB impls seems dodgy. I don't have confidence that you can run JAXB 2.x (included w/ JDK 8) and JAXB 3.x (added via dep) in the same JDK 8.

**Comment by ssill2** [comment]
> Running two JAXB impls seems dodgy. I don't have confidence that you can run JAXB 2.x (included w/ JDK 8) and JAXB 3.x (added via dep) in the same JDK 8.

I would agree.   Getting off Java8 first seems like a better idea.  Java11 really forces you to reevaluate what you bring in as deps, which is a really good exercise, even if it is a big pia.

**Comment by norrisjeremy** [comment]
> > > I've updated the release plans: v0.0.15.1 <-- test release to verify Maven Central accepts the release (in-progress 2022/04/21) v0.15.1 <-- Should be EXACTLY the same as v0.0.15.1 (this is to provide changes for folks waiting on unreleased changes that were targeted for v0.14.1) v2.3.6 <-- Should be EXACTLY the same as v0.15.1 (to match the JAXB version)
> > > I will then make a v2.x branch to maintain any needed jdk 8.x+jaxb 2.x changes going forward
> > > Then 'master' will be bumped to 'v3.x' (to match JAXB version) and converted to jdk 11+jakarta.
> > 
> > 
> > JAXB 3.x works fine with Java 8, I hope you won't force the 3.x update of this plugin require Java 11+?
> 
> Running two JAXB impls seems dodgy. I don't have confidence that you can run JAXB 2.x (included w/ JDK 8) and JAXB 3.x (added via dep) in the same JDK 8.

I can assure you it's perfectly fine, I have Java 8 projects built and running with JAXB 3.x, using the 3.x fork of this plugin from https://github.com/evolvedbinary/jvnet-jaxb-maven-plugin.

**Comment by norrisjeremy** [comment]
> > > JAXB 3.x works fine with Java 8, I hope you won't force the 3.x update of this plugin require Java 11+?
> > 
> > 
> > Running two JAXB impls seems dodgy. I don't have confidence that you can run JAXB 2.x (included w/ JDK 8) and JAXB 3.x (added via dep) in the same JDK 8.
> 
> I can assure you it's perfectly fine, I have Java 8 projects built and running with JAXB 3.x, using the 3.x fork of this plugin from https://github.com/evolvedbinary/jvnet-jaxb-maven-plugin.

I'll also point out that JAXB 3.x is part of the Jakarta EE 9.x spec, which supports Java 8.
It isn't until Jakarta EE 10.x that they have raised the minimum Java requirement to Java 11 and Jakarta EE 10.x will ship with JAXB 4.x.

**Comment by norrisjeremy** [comment]
> > I can assure you it's perfectly fine, I have Java 8 projects built and running with JAXB 3.x, using the 3.x fork of this plugin from https://github.com/evolvedbinary/jvnet-jaxb-maven-plugin.
> 
> I think that with java versions less than 9 allowed you to have two jars that provided the same class. in 9+ you'll get all kinds of issues trying to do that. Especially if you modularize your projects, i.e. add a module-info.java file to it. if you do, it will flat out fail to compile if two dependency jars provide the same class. I'm also using the evolvedbinary fork, but in my case it was to get compilation and functionality working on java17.

I think you are confused: JAXB 3.x uses `jakarta` package namespace, while JAXB 2.x uses `javax` package namespaces, thus there are no overlapping classes. It works perfectly fine.

**Comment by ssill2** [comment]
You're probably right, I've not tried to have two different versions.  You must have a very specialized use-case to need 2 different jaxb versions in the same codebase.   I remember how much of a pain it was going from 8->11 though.  That and the upgrade I just did to 17 made it obvious the most complicated voodoo in my build was the xjc schema generation lol.  I hate touching that stuff after I've gotten it working.

**Comment by norrisjeremy** [comment]
> You're probably right, I've not tried to have two different versions. You must have a very specialized use-case to need 2 different jaxb versions in the same codebase. I remember how much of a pain it was going from 8->11 though. That and the upgrade I just did to 17 made it obvious the most complicated voodoo in my build was the xjc schema generation lol. I hate touching that stuff after I've gotten it working.

I never said I use multiple JAXB runtimes in the same project: we upgraded our projects wholesale from JAXB 2.x to JAXB 3.x. They full use JAXB 3.x (`jakarta` namespace) and they build and run fine under both Java 8 and Java 11+.

My only point is that it would be shame for the 3.x version of this maven plugin, which I understand will target JAXB 3.x (`jakarta` package namespace only), to arbitrarily require Java 11+, even though JAXB 3.x fully supports Java 8.

**Comment by mattrpav** [comment]
@ssill2 my concern is on the impl side with the service activation and internals, but your point about JEE not moving to 11 gives me better confidence.

I'll plan to leave compiler and source set to 1.8 for now when I get to the 3.x effort.

**Comment by norrisjeremy** [comment]
> @ssill2 my concern is on the impl side with the service activation and internals, but your point about JEE not moving to 11 gives me better confidence.
> 
> I'll plan to leave compiler and source set to 1.8 for now when I get to the 3.x effort.

Perfect, thanks! :)

**Comment by mattrpav** [comment]
UPDATE: Release process looking good. Just waiting on OSSRH to grant permission to release to Maven Central.

**Comment by mattrpav** [comment]
OSSRH granted permission to repos. v0.0.15.1 has been released to Maven Central.

PLEASE test this for regression against your JDK 8 + javax projects.

**Comment by mattrpav** [comment]
New issue-- for the jakarta jaxb v3 migration we'll need to update the documentation to include two sections (javax vs jakarta).

If you have a couple spare hours to pitch in, this would be a big help!

**Comment by ssill2** [comment]
I can take a look.

**Comment by mattrpav** [comment]
v0.15.1 has been released to maven central

**Comment by mattrpav** [comment]
2022/04/25 update:

Updated releases have been pushed to Maven Central (0.0.15.1, 0.15.1). 

1. New jaxb maven plugin roadmap ticket #233
2. v2.3.6 will be released after some end-user verification (@ssill2). This version bump will over come the inadvertent v1.11.1 release of jaxb2-basics and provide alignment with that module.

Moving forward, please move discussions to the relevant GH issues and roadmap tickets.

Thank you everyone for your patience!

**Comment by ssill2** [comment]
thanks matt for all the work.  I'll try to test this in the next day or so.

**Comment by michael-o** [comment]
Holy moly, just found this after more than a year of not upgrding this plugin. I am shocked that he passed away. I was it touch with him 1:1 many times 8+ years ago regarding this plugin helping/testing. @mattrpav Thank you picking this up. Alexey's legacy should not vanish!
